### PR TITLE
[posix] disable addrconf generating LL address

### PR DESCRIPTION
**Pull request by bukepo**
This commit disables addrconf generating address by set addr_gen_mode to IN6_ADDR_GEN_MODE_NONE, otherwise linux may generate a link local address based on link type and presense of EUI64 address. This allows OpenThread to fully manage link local address of its network interface.

### CODE REVIEW

**Comment by bukepo** [comment]
This should address the native commissioner failure surfaced in #8526

**Review by wgtdkp — APPROVED** [review]
LGTM :+1:

**Comment by jwhui** [comment]
@bukepo , the clang build checks are failing:
```
/home/runner/work/openthread/openthread/src/posix/platform/netif.cpp:1654:25: error: implicit conversion changes signedness: 'unsigned int' to 'int' [-Werror,-Wsign-conversion]
    req.ifi.ifi_index = gNetifIndex;
                      ~ ^~~~~~~~~~~
1 error generated.
```

File: src/posix/platform/netif.cpp [hunk 1]
```diff
@@ -346,6 +351,8 @@ void AddRtAttr(struct nlmsghdr *aHeader, uint32_t aMaxLen, uint8_t aType, const
         memcpy(RTA_DATA(rta), aData, aLen);
     }
     aHeader->nlmsg_len = NLMSG_ALIGN(aHeader->nlmsg_len) + RTA_ALIGN(len);
+
+    return rta;
```
Comment (abtink) [inline src/posix/platform/netif.cpp:355]: I was wondering how do we return a pointer? Is it allocated somehow? Who owns it?

Looks like we cast the `aHeader`? 
```c++
rta  = (struct rtattr *)((char *)(aHeader) + NLMSG_ALIGN((aHeader)->nlmsg_len));
```

And `aHeader` always comes from `nh` in a:
```c++
    struct
    {
        struct nlmsghdr  nh;
        struct ifinfomsg ifi;
        char             buf[512];
    } req;
```

May make sense todefine this `struct` as a type and pass this instead? May be in a future PR?

**Review by abtink — APPROVED** [review]
LGTM. Thanks.
One idea for future above.

File: src/posix/platform/netif.cpp (hunk 1 above)
Comment (bukepo) [reply src/posix/platform/netif.cpp:355]: Yes, we may create a class to simplify generating netlink messages in a future PR.

**Comment by bukepo** [comment]
> @bukepo , the clang build checks are failing:
> 
> ```
> /home/runner/work/openthread/openthread/src/posix/platform/netif.cpp:1654:25: error: implicit conversion changes signedness: 'unsigned int' to 'int' [-Werror,-Wsign-conversion]
>     req.ifi.ifi_index = gNetifIndex;
>                       ~ ^~~~~~~~~~~
> 1 error generated.
> ```

just fixed the issue.

**Review by jwhui — APPROVED** [review]

**Pull request merged**